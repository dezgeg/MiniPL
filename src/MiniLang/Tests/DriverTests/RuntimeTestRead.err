Runtime error:
No input available for 'read' statement at line 25, columns 5-15
