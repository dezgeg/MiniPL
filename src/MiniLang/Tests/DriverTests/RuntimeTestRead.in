42
-100
asd

true
false