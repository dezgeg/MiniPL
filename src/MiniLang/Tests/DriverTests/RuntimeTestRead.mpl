print "Testing read statement... ";

var i : int;
var s : string;
var b : bool;

read i;
assert i = 42;
read i;
assert i = -100;

read s;
assert s = "asd";
read s;
assert s = "";

read b;
assert b;
read b;
assert !b;

print "OK so far\n";

var not_in_file : int;
read not_in_file;

// not reached
print "FAIL!\n";
assert false;