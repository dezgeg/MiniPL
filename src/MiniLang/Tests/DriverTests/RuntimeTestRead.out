Testing read statement... OK so far
